api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/calculation/pvls/BreastfeedingPregnantCalculationTest.java

```diff
@@ -0,0 +1,79 @@
+package org.openmrs.module.eptsreports.reporting.intergrated.calculation.pvls;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.api.context.Context;
+import org.openmrs.calculation.patient.PatientCalculation;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.reporting.calculation.pvls.BreastfeedingPregnantCalculation;
+import org.openmrs.module.eptsreports.reporting.intergrated.calculation.BasePatientCalculationTest;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportConstants.PregnantOrBreastfeedingWomen;
+
+public class BreastfeedingPregnantCalculationTest extends BasePatientCalculationTest {
+
+  @Override
+  public PatientCalculation getCalculation() {
+
+    return Context.getRegisteredComponents(BreastfeedingPregnantCalculation.class).get(0);
+  }
+
+  @Override
+  public Collection<Integer> getCohort() {
+    return Arrays.asList(new Integer[] {7, 8, 501});
+  }
+
+  @Override
+  public CalculationResultMap getResult() {
+    PatientCalculation calculation = getCalculation();
+    CalculationResultMap map = new CalculationResultMap();
+
+    PatientCalculationContext evaluationContext = getEvaluationContext();
+
+    // Patient marked as Pregnant in PTV program
+    map.put(501, new SimpleResult(true, calculation, evaluationContext));
+
+    // Patient is is marked as pregnant(YES) in Adult initial Followup
+    map.put(7, new SimpleResult(true, calculation, evaluationContext));
+
+    map.put(8, new SimpleResult(true, calculation, evaluationContext));
+
+    return map;
+  }
+
+  @Before
+  public void initialise() throws Exception {
+    executeDataSet("pvlsTest.xml");
+    params.put("state", PregnantOrBreastfeedingWomen.PREGNANTWOMEN);
+  }
+
+  @Test
+  public void shouldReturnPregnantIfMostRecent() {
+    setEvaluationContext(testsHelper.getDate("2018-09-20 00:00:00.0"));
+    params.put("state", PregnantOrBreastfeedingWomen.BREASTFEEDINGWOMEN);
```

Comment:
@ningosi shouldn't it be `PREGNANTWOMEN` here? Maybe this test belongs to `shouldReturnBreastfeedingIfMostRecent`.

Reply:
Sure, I will turn this around